client/extensions/woocommerce/app/products/product-form-variations-card.js

```diff
@@ -12,69 +12,121 @@ import ProductVariationTypesForm from './product-variation-types-form';
 import ProductFormVariationsTable from './product-form-variations-table';
 import FormToggle from 'components/forms/form-toggle';
 
-const ProductFormVariationsCard = ( {
-	product,
-	variations,
-	editProduct,
-	translate,
-	editProductAttribute,
-	editProductVariation
-} ) => {
-	// TODO When toggling between types, don't destory a user's previous edits. Save to component state and restore on toggle.
-	const handleToggle = () => {
-		if ( 'variable' !== product.type ) {
-			editProduct( product, { type: 'variable' } );
+class ProductFormVariationsCard extends Component {
+
+	state = {
+		simpleProduct: [],
+		variationAttributes: [],
+	};
+
+	static propTypes = {
+		product: PropTypes.shape( {
+			type: PropTypes.string.isRequired,
+			name: PropTypes.string,
+			attributes: PropTypes.array,
+		} ),
+		variations: PropTypes.array,
+		editProduct: PropTypes.func.isRequired,
+		editProductAttribute: PropTypes.func.isRequired,
+		editProductVariation: PropTypes.func.isRequired,
+	};
+
+	/*
+	 * When switching between product types, we clean up the Redux state to drop fields or variation attributes
+	 * that are no longer valid, so we can keep API calls clean. This is done in setVariable and setSimple.
+	 * To preserve a users edits, we store these changes in component state, and restore them if they toggle back.
+	 * A user shouldn't loose work just because they locally toggled a card.
+	 */
+	handleToggle = () => {
+		const simpleFields = [ 'dimensions', 'weight', 'regular_price', 'manage_stock', 'stock_quantity', 'backorders' ];
+		if ( 'variable' !== this.props.product.type ) {
+			this.setProductTypeVariable( simpleFields );
 		} else {
-			const attributes = ( product.attributes && product.attributes.filter( attribute => ! attribute.variation ) ) || null;
-			editProduct( product, { type: 'simple', attributes } );
+			this.setProductTypeSimple( simpleFields );
 		}
-	};
+	}
+
+	setProductTypeVariable( simpleFields ) {
```

Comment:
Hmm... I'm not sure if I agree.  A redux action is already used to actually _make_ the updates inside `setProductTypeVariable` -- I'm not sure the backup we are making needs to go somewhere else in Redux state too. What benefit do you see storing the backed-up fields there?

If the user never toggles back, we wouldn't do anything with that state and it would be another thing to clear out after save. It would also add another reducer and state sub-tree just to store back-up data that we only need on the current page load. I also don't see toggling back and forth as a common user action (except maybe when they are first learning their way around the UI) so I'm not sure if we should spend too much time on this here.

If people disagree, I can make changes. Or maybe it's something we could flag as a potential improvement for later?